static/app/views/automations/components/automationBuilderContext.tsx

```diff
@@ -395,6 +446,96 @@ function updateIfCondition(
   };
 }
 
+function addIfAction(
+  state: AutomationBuilderState,
+  action: AddIfActionAction
+): AutomationBuilderState {
+  const {groupIndex, actionType} = action;
+  return {
+    ...state,
+    actionFilters: state.actionFilters.map((group, i) => {
```

Comment:
cc @natemoo-re is form state all 1 large state? could we break it down into multiple contexts so this is a bit more predictable or is this so we can use the shard Form component?

Reply:
We definitely could break this down further! The goal of managing state in a single reducer was to keep everything directly in sync with the API's data structure so we don't have to maintain an additional mapping layer. We also get the added benefit of deriving the shared Form component's initial state from this one object.

When we discussed in DMs, I believe your suggestion was to break this down more so that each trigger / filter / action has its own individual that the rows hook into. That would definitely improve the readability here.

Reply:
Unfortunately I think that keeping form state as 1 state keeps things the "simplest" for now, since it would match the structure of the data that we are submitting to the API and it would be easier to use an existing automation's data to populate the form in the case of editing an automation.

I think breaking this down would make things more difficult to follow overall, since we would need to somehow manage each data condition group's individual state and keep track of how many data condition groups there are for action filters.

We do have contexts for each data condition to at keep each row self-contained.
https://github.com/getsentry/sentry/blob/82750d29864f782aab2c552dada38b59020d6f51/static/app/views/automations/components/dataConditionNodeList.tsx#L46-L54